.gitlab-ci.yml

```diff
@@ -1,5 +1,60 @@
-image: docker:dind
+image: docker:18-git
 
-deploy:
+stages:
+  - build
+
+before_script:
+  - export VERSION=$(cat VERSION)
+  - export CI_REGISTRY=${CI_REGISTRY:-hub.docker.com}
+  - export CI_REGISTRY_USER=${CI_REGISTRY_USER:-gitlab-ci-token}
+  - export CI_REGISTRY_PASS=${CI_REGISTRY_PASS:-${CI_JOB_TOKEN}}
+  - export DOCKER_IMAGE=${DOCKER_IMAGE:-${CI_REGISTRY}/${CI_PROJECT_PATH}}
+  - |
+    if [ "${DOCKER_IMAGE}" = "/" ]; then
+      export DOCKER_IMAGE=sameersbn/gitlab
+    fi
+
+docker:build:
+  stage: build
+  only:
+    - master
+  script:
+    - docker login -u ${CI_REGISTRY_USER} -p ${CI_REGISTRY_PASS} ${CI_REGISTRY}
+    - docker build
+      --pull
+      --cache-from=${DOCKER_IMAGE}
+      --build-arg=VCS_REF=$(git rev-parse --short HEAD)
+      --build-arg=BUILD_DATE="$(date +"%Y-%m-%d %H:%M:%S%:z")"
+      --build-arg=VERSION=${VERSION}
+      --tag ${DOCKER_IMAGE} .
+    - docker push ${DOCKER_IMAGE}
```

Comment:
@GJRTimmer Should this read `docker push ${DOCKER_IMAGE}:latest`?
 
Otherwise it pushes *all* of local tags found, doesn't it?

/cc @solidnerd

Reply:
So, it'd read:

```diff
diff --git a/.gitlab-ci.yml b/.gitlab-ci.yml
index 6e8d57a..8cb54a8 100644
--- a/.gitlab-ci.yml
+++ b/.gitlab-ci.yml
@@ -26,8 +26,8 @@ docker:build:
       --build-arg=VCS_REF=$(git rev-parse --short HEAD)
       --build-arg=BUILD_DATE="$(date +"%Y-%m-%d %H:%M:%S%:z")"
       --build-arg=VERSION=${VERSION}
-      --tag ${DOCKER_IMAGE} .
-    - docker push ${DOCKER_IMAGE}
+      --tag ${DOCKER_IMAGE}:${VCS_REF} .
+    - docker push ${DOCKER_IMAGE}:${VCS_REF}
 
 docker:build:branches:
   stage: build
```